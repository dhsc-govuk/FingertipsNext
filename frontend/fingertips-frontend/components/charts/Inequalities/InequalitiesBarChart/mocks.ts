import { InequalitiesBarChartData } from '@/components/charts/Inequalities/helpers/inequalitiesHelpers';

export const getTestData = (): InequalitiesBarChartData => ({
  areaCode: 'A1425',
  areaName: 'South FooBar',
  data: {
    period: 2008,
    inequalities: {
      Male: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: false,
      },
      Persons: {
        value: 135.149304,
        count: 222,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: true,
      },
      Female: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: false,
      },
    },
  },
});
