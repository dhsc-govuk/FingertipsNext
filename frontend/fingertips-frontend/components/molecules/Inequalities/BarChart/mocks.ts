import { InequalitiesBarChartData } from '@/components/organisms/Inequalities/inequalitiesHelpers';

export const getTestData = (): InequalitiesBarChartData => ({
  areaName: 'South FooBar',
  data: {
    period: 2008,
    inequalities: {
      Male: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
      },
      Persons: {
        value: 135.149304,
        count: 222,
        upper: 578.32766,
        lower: 441.69151,
      },
      Female: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
      },
    },
  },
});
