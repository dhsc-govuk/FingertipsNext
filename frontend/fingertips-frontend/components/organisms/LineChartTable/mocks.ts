export const MOCK_ENGLAND_DATA = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      year: 2004,
      count: 200,
      value: 904.874,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '0-4',
      sex: 'Female',
    },
    {
      year: 2008,
      count: 500,
      value: 965.9843,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '10-14',
      sex: 'Female',
    },
  ],
};

export const MOCK_HEALTH_DATA = [
  {
    areaCode: 'A1425',
    areaName: 'Greater Manchester ICB - 00T',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 890.305692,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
    ],
  },
  {
    ...MOCK_ENGLAND_DATA,
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        year: 2004,
        count: 222,
        value: 135.149304,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
    ],
  },
];
