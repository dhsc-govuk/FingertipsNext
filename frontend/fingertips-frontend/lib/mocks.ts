import {
  AgeData,
  DeprivationData,
  HealthDataPoint,
  HealthDataPointTrendEnum,
  SexData,
} from '@/generated-sources/ft-api-client';
import { mockDatePeriod } from '@/mock/data/mockDatePeriod';

export const noDeprivation: DeprivationData = {
  sequence: 1,
  type: 'All',
  value: 'All',
  isAggregate: true,
};

export const personsSex: SexData = {
  value: 'Persons',
  isAggregate: true,
};

export const maleSex: SexData = {
  value: 'Male',
  isAggregate: false,
};

export const femaleSex: SexData = {
  value: 'Female',
  isAggregate: false,
};

export const allAgesAge: AgeData = {
  value: 'All ages',
  isAggregate: true,
};

export const disaggregatedAge = (age: string) => {
  return {
    value: age,
    isAggregate: false,
  };
};

export const healthDataPoint: HealthDataPoint = {
  count: 389,
  lowerCi: 441.69151,
  upperCi: 578.32766,
  value: 278.29134,
  datePeriod: mockDatePeriod(2006),
  sex: personsSex,
  ageBand: allAgesAge,
  trend: HealthDataPointTrendEnum.NotYetCalculated,
  deprivation: noDeprivation,
};
