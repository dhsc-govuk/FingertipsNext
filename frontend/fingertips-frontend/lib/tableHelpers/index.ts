export enum LineChartTableHeadingEnum {
  AreaPeriod = 'Period',
  BenchmarkTrend = 'Compared to benchmark',
  AreaCount = 'Count',
  AreaValue = 'Value',
  AreaLower = 'Lower',
  AreaUpper = 'Upper',
  BenchmarkValue = 'Value ',
}

export enum InequalitiesSexTableHeadingsEnum {
  PERIOD = 'Period',
  PERSONS = 'Persons',
  MALE = 'Male',
  FEMALE = 'Female',
  BENCHMARK = 'England benchmark',
}

export interface LineChartTableRowData {
  period: number;
  count: number;
  value: number;
  lower: number;
  upper: number;
}

export interface InequalitiesSexTableRowData {
  period: number;
  persons: number | undefined;
  male: number | undefined;
  female: number | undefined;
  englandBenchmark: number | undefined;
}

export const convertToPercentage = (value: number): string => {
  // dummy function to do percentage conversions until real conversion logic is provided
  return `${((value / 10000) * 100).toFixed(1)}%`;
};
