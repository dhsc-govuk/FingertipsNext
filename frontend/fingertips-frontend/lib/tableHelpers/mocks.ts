export const MOCK_ENGLAND_DATA = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      year: 2004,
      count: 200,
      value: 904.874,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '0-4',
      sex: 'All',
    },
    {
      year: 2008,
      count: 500,
      value: 965.9843,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '10-14',
      sex: 'All',
    },
  ],
};

export const MOCK_HEALTH_DATA = [
  {
    areaCode: 'A1425',
    areaName: 'Greater Manchester ICB - 00T',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 890.305692,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'All',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'All',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Male',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Female',
      },
    ],
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 135.149304,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'All',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Male',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Female',
      },
    ],
  },
];

export const MOCK_PARENT_DATA = {
  areaCode: 'E40000014',
  areaName: 'NHS North West Region',
  healthData: [
    {
      year: 2008,
      count: 777,
      value: 1000,
      lowerCi: 500,
      upperCi: 1500,
      ageBand: 'All',
      sex: 'All',
    },
    {
      year: 2004,
      count: 777,
      value: 1000,
      lowerCi: 500,
      upperCi: 1500,
      ageBand: 'All',
      sex: 'All',
    },
  ],
};
