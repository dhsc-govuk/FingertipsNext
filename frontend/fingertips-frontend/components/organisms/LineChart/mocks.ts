import {
  HealthDataForArea,
  HealthDataPointTrendEnum,
} from '@/generated-sources/ft-api-client';
import { allAgesAge, noDeprivation, personsSex } from '@/lib/mocks';

export const mockIndicatorData: HealthDataForArea[] = [
  {
    areaCode: 'A1425',
    areaName: 'North FooBar',
    healthData: [
      {
        count: 389,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        value: 278.29134,
        year: 2006,
        periodLabel: '2006',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 267,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        value: 703.420759,
        year: 2004,
        periodLabel: '2004',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        count: 654,
        lowerCi: 750.69151,
        upperCi: 800.32766,
        value: 786.27434,
        year: 2010,
        periodLabel: '2010',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 567,
        lowerCi: 440.69151,
        upperCi: 420.32766,
        value: 435.420759,
        year: 2007,
        periodLabel: '2007',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
  {
    areaCode: 'A1427',
    areaName: 'East FooBar',
    healthData: [
      {
        count: 953,
        lowerCi: 460.69151,
        upperCi: 500.32766,
        value: 478.27434,
        year: 2020,
        periodLabel: '2020',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 567,
        lowerCi: 220.69151,
        upperCi: 250.32766,
        value: 234.420759,
        year: 2012,
        periodLabel: '2012',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
];

export const mockBenchmarkData: HealthDataForArea = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      count: 389,
      lowerCi: 441.69151,
      upperCi: 578.32766,
      value: 278.29134,
      year: 2006,
      periodLabel: '2006',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
    {
      count: 267,
      lowerCi: 441.69151,
      upperCi: 578.32766,
      value: 703.420759,
      year: 2004,
      periodLabel: '2004',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
  ],
};

export const mockParentData: HealthDataForArea = {
  areaCode: 'P001',
  areaName: 'Parent',
  healthData: [
    {
      count: 100,
      lowerCi: 200,
      upperCi: 400,
      value: 300,
      year: 2006,
      periodLabel: '2006',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
    {
      count: 101,
      lowerCi: 201,
      upperCi: 401,
      value: 301,
      year: 2004,
      periodLabel: '2004',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
  ],
};
