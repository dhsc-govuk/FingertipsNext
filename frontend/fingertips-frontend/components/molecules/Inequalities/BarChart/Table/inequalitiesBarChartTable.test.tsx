import { InequalitiesBarChartTable, InequalitiesBarChartTableHeaders } from '.';
import { GROUPED_YEAR_DATA, MOCK_HEALTH_DATA } from '@/lib/tableHelpers/mocks';
import { render, screen } from '@testing-library/react';
import { expect } from '@jest/globals';
import {
  Inequalities,
  InequalitiesBarChartTableData,
} from '@/components/organisms/Inequalities/inequalitiesHelpers';

describe('Inequalities bar chart table suite', () => {
  describe('Sex inequality', () => {
    const tableData: InequalitiesBarChartTableData = {
      areaName: MOCK_HEALTH_DATA[1].areaName,
      data: {
        period: 2008,
        inequalities: {
          Persons: {
            value: 135.149304,
            count: 222,
            upper: 578.32766,
            lower: 441.69151,
          },
          Male: {
            value: 890.328253,
            count: 131,
            upper: 578.32766,
            lower: 441.69151,
          },
          Female: {
            value: 890.328253,
            count: 131,
            upper: 578.32766,
            lower: 441.69151,
          },
        },
      },
    };

    it('should render the inequalitiesBarChartTable component', () => {
      render(
        <InequalitiesBarChartTable
          tableData={tableData}
          yearlyHealthDataGroupedByInequalities={GROUPED_YEAR_DATA}
          type={Inequalities.Sex}
        />
      );

      expect(
        screen.getByTestId('inequalitiesBarChartTable-component')
      ).toBeInTheDocument();
    });

    it('should render expected elements', () => {
      const expectedGroupings = ['Persons', 'Male', 'Female'];

      render(
        <InequalitiesBarChartTable
          tableData={tableData}
          yearlyHealthDataGroupedByInequalities={GROUPED_YEAR_DATA}
          type={Inequalities.Sex}
        />
      );

      expect(screen.getByRole('table')).toBeInTheDocument();
      Object.values(InequalitiesBarChartTableHeaders).forEach((header) => {
        expect(screen.getByTestId(`heading-${header}`)).toBeInTheDocument();
      });
      expectedGroupings.forEach((grouping) => {
        expect(screen.getByText(grouping)).toBeInTheDocument();
      });
    });

    it('should display x if data point is not available', () => {
      const expectedNumberOfRows = 3;
      const mockData: InequalitiesBarChartTableData = {
        areaName: tableData.areaName,
        data: {
          period: 2004,
          inequalities: {
            Persons: { value: 890.3432 },
            Male: { value: 703.420759 },
            Female: { value: 703.420759 },
          },
        },
      };

      render(
        <InequalitiesBarChartTable
          tableData={mockData}
          yearlyHealthDataGroupedByInequalities={GROUPED_YEAR_DATA}
          type={Inequalities.Sex}
        />
      );
      expect(screen.getAllByRole('cell')).toHaveLength(
        expectedNumberOfRows *
          Object.values(InequalitiesBarChartTableHeaders).length
      );
      screen
        .getAllByTestId('not-available')
        .forEach((id) => expect(id).toHaveTextContent('X'));
    });

    it('check if the measurementUnit value "kg" is rendered correctly with braces', () => {
      render(
        <InequalitiesBarChartTable
          tableData={tableData}
          yearlyHealthDataGroupedByInequalities={GROUPED_YEAR_DATA}
          type={Inequalities.Sex}
          measurementUnit="kg"
        />
      );
      expect(screen.getByText('(kg)')).toBeInTheDocument();
    });

    it('snapshot test - should match snapshot', () => {
      const container = render(
        <InequalitiesBarChartTable
          tableData={tableData}
          yearlyHealthDataGroupedByInequalities={GROUPED_YEAR_DATA}
          type={Inequalities.Sex}
        />
      );

      expect(container.asFragment()).toMatchSnapshot();
    });
  });
});
