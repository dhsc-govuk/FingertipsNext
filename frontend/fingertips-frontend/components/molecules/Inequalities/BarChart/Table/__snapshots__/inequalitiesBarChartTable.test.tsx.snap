// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inequalities bar chart table suite > Sex inequality > snapshot test - should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="inequalitiesBarChartTable-component"
    id="inequalitiesBarChartTable"
  >
    <h4
      class="heading__StyledHeading-sc-1dsobun-0 EoZyU sc-jxOSlx bLSbGR"
    >
      inequalities for South FooBar, 2008
    </h4>
    <table
      class="src__StyledTable-sc-16s660v-0 duOufa"
    >
      <thead
        class="TableHeader-sc-1n1nw4b-0"
      >
        <tr
          class="Row-sc-1y8eqfp-0"
        >
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-gEvEer sc-eDPEul hoeTcz dzKXJe"
            colspan="4"
          />
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-gEvEer sc-eDPEul sc-esYiGF hoeTcz dzKXJe fPChvl"
            colspan="2"
            style="text-align: center;"
          >
            95%
            <br />
            confidence
            <br />
            limits
          </th>
        </tr>
        <tr
          class="Row-sc-1y8eqfp-0"
        >
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-gEvEer sc-eDPEul hoeTcz dzKXJe"
            data-testid="heading-Inequality type"
            style="width: 16%;"
          >
            Inequality type
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-cwHptR kxOGSV"
            data-testid="heading-Compared to persons"
            style="width: 16%;"
          >
            Compared to persons
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-gEvEer sc-eqUAAy hoeTcz brMjvX"
            data-testid="heading-Count"
            style="width: 16%;"
          >
            Count
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-gEvEer sc-eqUAAy hoeTcz brMjvX"
            data-testid="heading-Value"
            style="width: 16%;"
          >
            Value
            <span
              data-testid="inequalitiesBarChart-measurementUnit"
              style="display: block; margin: auto;"
            >
               kg
            </span>
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-gEvEer sc-eqUAAy hoeTcz brMjvX"
            data-testid="heading-Lower"
            style="width: 16%;"
          >
            Lower
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-gEvEer sc-eqUAAy hoeTcz brMjvX"
            data-testid="heading-Upper"
            style="width: 16%; padding-right: 10px;"
          >
            Upper
          </th>
        </tr>
      </thead>
      <tbody
        class="TableBody-sc-1qqarm8-0"
      >
        <tr
          class="Row-sc-1y8eqfp-0"
        >
          <td
            class="Cell-sc-msv0l6-0 sc-cPiKLX gaSlHc hKweSg"
          >
            Persons
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-dAlyuH gaSlHc kRfgxG"
          />
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-gsFSXq gaSlHc eXbeNP dLcQtJ"
          >
            222
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-gsFSXq gaSlHc eXbeNP dLcQtJ"
          >
            135.1
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-gsFSXq gaSlHc eXbeNP dLcQtJ"
          >
            441.7
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-dLMFU gaSlHc eXbeNP jlNuyg"
          >
            578.3
          </td>
        </tr>
        <tr
          class="Row-sc-1y8eqfp-0"
        >
          <td
            class="Cell-sc-msv0l6-0 sc-cPiKLX gaSlHc hKweSg"
          >
            Male
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-dAlyuH gaSlHc kRfgxG"
          />
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-gsFSXq gaSlHc eXbeNP dLcQtJ"
          >
            131
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-gsFSXq gaSlHc eXbeNP dLcQtJ"
          >
            890.3
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-gsFSXq gaSlHc eXbeNP dLcQtJ"
          >
            441.7
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-dLMFU gaSlHc eXbeNP jlNuyg"
          >
            578.3
          </td>
        </tr>
        <tr
          class="Row-sc-1y8eqfp-0"
        >
          <td
            class="Cell-sc-msv0l6-0 sc-cPiKLX gaSlHc hKweSg"
          >
            Female
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-dAlyuH gaSlHc kRfgxG"
          />
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-gsFSXq gaSlHc eXbeNP dLcQtJ"
          >
            131
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-gsFSXq gaSlHc eXbeNP dLcQtJ"
          >
            890.3
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-gsFSXq gaSlHc eXbeNP dLcQtJ"
          >
            441.7
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-iGgWBj sc-dLMFU gaSlHc eXbeNP jlNuyg"
          >
            578.3
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="exportOnly"
      style="display: none;"
    >
      <div
        class="sc-hzhJZQ fCBUdc"
      >
        Â©Crown copyright 2024. Office for Health Improvement & Disparities.
        <br />
        Public Health Profiles accessed on 25 December 2024 www.fingertips.phe.org.uk
      </div>
    </div>
  </div>
  <button
    class="src__StyledButton-sc-19ocyxv-0 eEShVH sc-koXPp cbdPWW"
    data-testid="inequalitiesBarChartTable-export-button"
  >
    Export options
  </button>
</DocumentFragment>
`;

exports[`Inequalities bar chart table suite Sex inequality snapshot test - should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-top: 0;
  margin-bottom: 15px;
}

.c3 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
  font-weight: 700;
}

.c3:last-child {
  padding-right: 0;
}

.c9 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
}

.c9:last-child {
  padding-right: 0;
}

.c2 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  margin-bottom: 20px;
}

.c4 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  font-weight: bold;
  padding: 0.625em 0;
}

.c8 {
  text-align: right;
  padding-right: 10px;
  vertical-align: top;
}

.c12 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  padding-right: 0;
}

.c13 {
  text-align: right;
  padding-right: 10px;
}

.c11 {
  text-align: center;
  padding-left: 10px;
  padding-right: 10px;
}

.c7 {
  text-align: center;
  width: 10%;
  padding-left: 20px;
}

.c10 {
  width: 10%;
  padding-left: 10px;
}

.c14 {
  text-align: right;
  padding-right: 10px !important;
}

.c5 {
  text-align: left;
  vertical-align: top;
  padding-left: 10px;
}

.c15 {
  font-size: 14px;
  margin-bottom: 8px;
}

.c1 {
  font-size: 19px;
  margin: 0 0 30px 0;
}

.c6 {
  padding: 0.5em;
  text-align: center;
  vertical-align: top;
  white-space: nowrap;
}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {
  .c0 {
    color: #000;
  }
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 20px;
  }
}

@media print {
  .c2 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c2 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    margin-bottom: 30px;
  }
}

@media print {
  .c4 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c12 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c12 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

<div
    data-testid="inequalitiesBarChartTable-component"
    id="inequalitiesBarChartTable"
  >
    <h4
      class="c0 c1"
    >
      inequalities for South FooBar, 2008
    </h4>
    <table
      class="c2"
    >
      <thead
        class=""
      >
        <tr
          class=""
        >
          <th
            class="c3 c4 c5"
            colspan="4"
          />
          <th
            class="c3 c4 c5 c6"
            colspan="2"
            style="text-align: center;"
          >
            95%
            <br />
            confidence
            <br />
            limits
          </th>
        </tr>
        <tr
          class=""
        >
          <th
            class="c3 c4 c5"
            data-testid="heading-Inequality type"
            style="width: 16%;"
          >
            Inequality type
          </th>
          <th
            class="c3 c7"
            data-testid="heading-Compared to persons"
            style="width: 16%;"
          >
            Compared to persons
          </th>
          <th
            class="c3 c4 c8"
            data-testid="heading-Count"
            style="width: 16%;"
          >
            Count
          </th>
          <th
            class="c3 c4 c8"
            data-testid="heading-Value"
            style="width: 16%;"
          >
            Value
            <span
              data-testid="inequalitiesBarChart-measurementUnit"
              style="display: block; margin: auto;"
            >
               kg
            </span>
          </th>
          <th
            class="c3 c4 c8"
            data-testid="heading-Lower"
            style="width: 16%;"
          >
            Lower
          </th>
          <th
            class="c3 c4 c8"
            data-testid="heading-Upper"
            style="width: 16%; padding-right: 10px;"
          >
            Upper
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <td
            class="5 c9 c10"
          >
            Persons
          </td>
          <td
            class="1 c9 c11"
          />
          <td
            class="c9 c12 c13"
          >
            222
          </td>
          <td
            class="c9 c12 c13"
          >
            135.1
          </td>
          <td
            class="c9 c12 c13"
          >
            441.7
          </td>
          <td
            class="6 c9 c12 c14"
          >
            578.3
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class="5 c9 c10"
          >
            Male
          </td>
          <td
            class="1 c9 c11"
          />
          <td
            class="c9 c12 c13"
          >
            131
          </td>
          <td
            class="c9 c12 c13"
          >
            890.3
          </td>
          <td
            class="c9 c12 c13"
          >
            441.7
          </td>
          <td
            class="6 c9 c12 c14"
          >
            578.3
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class="5 c9 c10"
          >
            Female
          </td>
          <td
            class="1 c9 c11"
          />
          <td
            class="c9 c12 c13"
          >
            131
          </td>
          <td
            class="c9 c12 c13"
          >
            890.3
          </td>
          <td
            class="c9 c12 c13"
          >
            441.7
          </td>
          <td
            class="6 c9 c12 c14"
          >
            578.3
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="exportOnly"
      style="display: none;"
    >
      <div
        class="c15"
      >
        Â©Crown copyright 2024. Office for Health Improvement & Disparities.
        <br />
        Public Health Profiles accessed on 25 December 2024 www.fingertips.phe.org.uk
      </div>
    </div>
  </div>
  .c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 19px;
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  width: 100%;
  margin-top: 0;
  padding: 7px 10px;
  border: 2px solid transparent;
  border-radius: 0;
  color: #ffffff;
  background-color: #00703c;
  box-shadow: 0 2px 0 #002413;
  text-align: center;
  vertical-align: top;
  cursor: pointer;
  -webkit-appearance: none;
  margin-bottom: 22px;
}

.c0:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c0:link,
.c0:visited,
.c0:active,
.c0:hover {
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0::-moz-focus-inner {
  padding: 0;
  border: 0;
}

.c0:hover,
.c0:focus {
  background-color: #00572e;
}

.c0:active {
  top: 2px;
  box-shadow: none;
}

.c0::before {
  content: "";
  display: block;
  position: absolute;
  top: -2px;
  right: -2px;
  bottom: -4px;
  left: -2px;
  background: transparent;
}

.c0:active::before {
  top: -4px;
}

.c0:disabled {
  opacity: 0.5;
  background: #00703c;
}

.c0:disabled:hover {
  background-color: #00703c;
  cursor: default;
}

.c0:disabled:focus {
  outline: none;
}

.c0:disabled:active {
  top: 0;
  box-shadow: 0 2px 0 #002413;
}

.c0 svg {
  max-width: 15px;
}

.c1 {
  margin-top: 1rem;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    width: auto;
  }
}

@media only screen and (min-width:641px) {
  .c0 svg {
    margin-left: 10px;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 32px;
  }
}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

<button
    class="c0 c1"
    data-testid="inequalitiesBarChartTable-export-button"
  >
    Export options
  </button>
</DocumentFragment>
`;
